front/components/assistant_builder/submitAssistantBuilderForm.ts

```diff
@@ -120,7 +121,19 @@ export async function submitAssistantBuilderForm({
             type: "tables_query_configuration",
             name: a.name,
             description: a.description,
-            tables: Object.values(a.configuration),
+            tables: Object.values(a.configuration).flatMap(
+              ({ dataSourceView, selectedResources }) => {
+                return selectedResources.map((resource) => ({
+                  dataSourceId: dataSourceView.dataSource.name,
+                  dataSourceViewId: dataSourceView.sId,
+                  workspaceId: owner.sId,
```

Comment:
nit: Can we avoid using `owner` from the outside scope ?

Reply:
Given that we need the `sId` we can't really use something else.